R/cost_uplift.R

```diff
@@ -0,0 +1,78 @@
+#' Uplift costs
+#'
+#' @param data episode data
+#' @param uplift
+#'
+#' @return episode data with uplifted costs
+#' @export
+cost_uplift <- function(data) {
+  data <- data %>%
+    # attach a uplift scale as the last column
+    uplift_set() %>%
+    dplyr::mutate(
+      cost_total_net = cost_total_net * uplift,
+      cost_total_net_incdnas = cost_total_net_incdnas * uplift,
+      apr_cost = apr_cost * uplift,
+      may_cost = may_cost * uplift,
+      jun_cost = jun_cost * uplift,
+      jul_cost = jul_cost * uplift,
+      aug_cost = aug_cost * uplift,
+      sep_cost = sep_cost * uplift,
+      oct_cost = oct_cost * uplift,
+      nov_cost = nov_cost * uplift,
+      dec_cost = dec_cost * uplift,
+      jan_cost = jan_cost * uplift,
+      feb_cost = feb_cost * uplift,
+      mar_cost = mar_cost * uplift
+    ) %>%
+    # remove the last uplift column
+    dplyr::select(-uplift)
+  return(data)
+}
+
+#' set uplift scale
+#'
+#' @param data episode data
+#' @param latest_cost_year default financial year '2223'
+#'
+#' @return episode data with a uplift scale
+#' @export
+uplift_set <- function(data, latest_cost_year = "2223") {
+  # We have set uplifts to use for 2020/21, 2021/22 and 2022/23, provided by Paul Leak.
+  # For older years, don't uplift.
+  # For years after 2022/23 uplift by an additional 1% per year after the latest cost year (2022/23)
+  # For non plics recids use uplift of 1 so we won't change anything.
+
+  # to accelerate, create a data frame of year and uplift for match-joining
+  start_year <- 10
+  end_year <- 70
+  year <- paste0(start_year:end_year, (start_year + 1):(end_year + 1)) %>% as.numeric()
+  uplift_df <- data.frame(year, uplift = 1) %>%
```

Comment:
Use tibble so we avoid strings turning into factors.
```suggestion
  uplift_df <- tibble::tibble(year, uplift = 1) %>%
```